.github/scripts/build-tools/compact_data.py

```diff
@@ -0,0 +1,86 @@
+import json
+import os
+import re
+import sys
+
+class DataProcessor:
+    def __init__(self, data, dataDir):
+        self.data = data
+        self._dataDir = dataDir
+
+    _routeKeys = [ 'md', 'kml' ]
+    def _routes(self, data = None):
+        if data is None:
+            data = self.data
+
+        if isinstance(data, list):
+            for value in data:
+                yield from self._routes(value)
+        elif isinstance(data, dict):
+            for key, value in data.items():
+                if key in self._routeKeys:
+                    yield data
+                    break
+                else:
+                    yield from self._routes(value)

```

Comment:
this loop won't necessarily do what u expect imo, at least I assume your goal is to return the whole "dict" if we have the `_routeKeys` present, and only try to use the "value" if both of those properties are missing

the problem is that we might encounter the first property that isn't md, or kml, so we yield with the else branch, but in reality we might have those properties

shouldn't this be:
```python
if (data.keys() & self._routeKeys)):
  yield data
else:
  yield from self._routes(value)
```

Reply:
Ah, good catch! It was supposed to be a for..else... The indentation got goopy.

Reply:
It works correctly. The example code would have to be run inside the for-loop to cover all values. Running intersection on all keys and route keys for EVERY key in a loop seems unnecessary. O(N^2) while mine is O(N)